kotlin/src/LogBeat.kt

```diff
@@ -0,0 +1,64 @@
+// Copyright 2018 Sourcerer Inc. All Rights Reserved.
+// Author: Alexander Surkov (alex@sourcerer.io)
+
+package sourcererio.vacuumanalytics
+
+import com.github.kittinunf.fuel.core.FuelError
+import com.github.kittinunf.fuel.core.FuelManager
+import com.github.kittinunf.fuel.core.Method
+import com.github.kittinunf.fuel.core.Request
+import com.github.kittinunf.fuel.core.Response
+import java.net.HttpCookie
+
+/**
+ * Elastic LogStash logging.
+ */
+class LogBeat {
+    private var session: String = ""
+    private val fuelManager = FuelManager()
+
+    /**
+     * Creates a LogBeat instance.
+     *
+     * @param path [in] path to send logs to,
+     * @param res  [in] HTTP response of successful user authorization,
+     * @param sid  [in] session id of the HTTP response.
+     */
+    constructor(path: String, res: Response, sid: String) {
+        res.headers["Set-Cookie"]?.
+            flatMap { HttpCookie.parse(it) }?.
+            find { it.name == sid }?.
+            let { session = it.value }
+
+        fuelManager.basePath = path
+    }
+
+    /**
+     * Logs error.
+     */
+    fun error(message: String) {
+        beat("error", message)
+    }
+
+    /**
+     * Logs a beat.
+     */
+    fun beat(type: String, message: String) {
+        fuelManager.
+            request(
+                Method.POST,
+                "",
+                listOf(
+                    "session" to session,
```

Comment:
Usually, things you want to record with with an event (some of them are web only, some of the app only):

event type
event name
event arguments (name->value) pairs
anonymous user identity
named user identity
local timestamp
originator url
referrer url
user agent
platform: desktop/mobile/tablet
app version
@anatolykislov @makrusak Thoughts on what else?

Reply:
I wanted to start from something minimal, since I'm not yet clear on all use cases we want to handle, and then I was planning to extend the API in the integration stage. Having said that if you feel strong about those items, then there's not problem to add those from the start. However I'm not sure how certain items like user agent/orinator url and others are applied to the kotlin library.